sklearn/neighbors/base.py

```diff
@@ -773,11 +773,20 @@ def fit(self, X, y):
             self.outputs_2d_ = True
 
         check_classification_targets(y)
+
+        self._issparsemultilabel = issparse(y) and self.outputs_2d_
```

Comment:
Yes, we could store that in this way. Does save the need of using another variable.